### DISCUSSION TITLE
visual drill down

### DISCUSSION THREAD

**Discussion by yosiasz** [grafana/grafana-infinity-datasource#350] [Ideas]
Wish list.

for those faint of heart using UQL, this plugin would take it to another level if it had a visual drill down method. once you plug in an inline json or json api, it would parse the result and present you a visual json path you can use to drill down to what you want.

Thanks

**Comment by yesoreyeram** [comment]
Hi @yosiasz - Can you elaborate what do you mean by visual drill down and throw some examples of what you expect?

**Reply by yosiasz** [reply]
hello @yesoreyeram 

<img width="149" alt="image" src="https://user-images.githubusercontent.com/900951/177371418-0afa8f26-e30f-4d28-b4d3-a6306818cdd4.png">

This is from notepad++, so something visual especially for beginner users of your awesome plugin.

From 1 of your biggest fans. And once user selects the specific node and specific elements, uql is autogenerated or suggested.
Sort of like like a UQL wizard

**Reply by yesoreyeram** [reply]
Ah I see. You meant some sort of UQL query builder from the response. Definitely that is a long term goal.
